pkg/xds/secrets/ca_provider.go

```diff
@@ -31,12 +48,27 @@ func (s *meshCaProvider) Get(ctx context.Context, mesh *core_mesh.MeshResource)
 		return nil, nil, errors.New("CA backend is nil")
 	}
 
+	var cancel context.CancelFunc
+	timeout := backend.GetRootChain().GetRequestTimeout()
+	if timeout != nil {
+		ctx, cancel = context.WithTimeout(ctx, timeout.AsDuration())
+		defer cancel()
+	}
+
 	caManager, exist := s.caManagers[backend.Type]
 	if !exist {
 		return nil, nil, errors.Errorf("CA manager of type %s not exist", backend.Type)
 	}
 
-	certs, err := caManager.GetRootCert(ctx, mesh.GetMeta().GetName(), backend)
+	var certs [][]byte
+	var err error
+	func() {
+		start := time.Now()
+		defer func() {
+			s.latencyMetrics[backend.Type].WithLabelValues(backend.GetName()).Observe(float64(time.Since(start).Milliseconds()))
```

Comment:
Is this for if the function panics?

Reply:
@michaelbeaumont Yeah, it will always record how long this func takes, even if we add extra returns or it panics or something. We do this all over the place, eg `pkg/metrics/store/store.go` , `registerDNSHandler()` , `dataplane_watchdog_factory.go`. Mostly following those examples.

Reply:
Yeah, makes sense